File: Python/binary_search.py
def binary_search(list, target):
    first = 0
    last = len(list)-1
    while first <= last:
        midpoint = (first + last )//2
        
        if list[midpoint] == target:
            return midpoint
        elif list[midpoint] < target:
            first = midpoint + 1
        else:
            last = midpoint - 1
    return None
            
def verify(index):
    if index is not None:
        print("The item is in the index: ", index)
    else:
        print("The item is not in the index")
        
        
numbers = [-2,-1,0,1,2,3,4,5,6,7,8,9,10]
result = binary_search(numbers, 7)
verify(result)

File: JAVA/BinarySearch.java

public class BinarySearch {
    public static void main(String[] args) {
        int[] arr ={ 10,20,30,40,50,60,70,80,90,100 };
        int target = 41;
        int start = 0 , end = arr.length-1,done=0;
        while(start <= end){
            // (start + end ) / 2 ; start + end may exceed int range in java
            int mid = start + (end-start)/2;
            if (target>arr[mid]){
                start = mid + 1;
            }
            else if (target<arr[mid]){
                end = mid - 1;
            }
            else{
                System.out.println("Target Found at index "+(mid+1));
                done = 1;
                break;
            }

        }
        if(done == 0){
            System.out.println("Target Not Found in the Array");
        }
    }
}
